### PR TITLE
Add a planar-shadow pass to the built-in effect

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

File: cocos/render-scene/scene/submodel.ts [hunk 1]
```diff
@@ -169,22 +167,6 @@ export class SubModel {
...
-        return this._planarInstanceShader;
-    }
-
-    /**
-     * @en The shader for rendering the planar shadow.
-     * @zh 用于渲染平面阴影的着色器。
-     */
-    get planarShader (): Shader | null {
-        return this._planarShader;
-    }
```
Comment (troublemaker52025) [inline cocos/render-scene/scene/submodel.ts:186]: Delete trick practice，pass & shader is built in by effect

**Review by jk20012001 — APPROVED** [review]

File: cocos/rendering/planar-shadow-queue.ts [hunk 2]
```diff
@@ -45,12 +61,22 @@ export class PlanarShadowQueue {
         this._pipeline = pipeline;
     }
 
+    /**
+     * @zh
+     * clear light-Batched-Queue
+     */
```
Comment (minggo) [inline cocos/rendering/planar-shadow-queue.ts:69]: Error comment. There is not comment for `@zh`.

File: cocos/rendering/planar-shadow-queue.ts [hunk 3]
```diff
@@ -59,17 +85,15 @@ export class PlanarShadowQueue {
...
 
         const models = scene.models;
+        const visibility = camera.visibility;
         for (let i = 0; i < models.length; i++) {
             const model = models[i];
             if (scene.isCulledByLod(camera, model)) {
                 continue;
             }
-            if (model.enabled && model.node && model.castShadow) { this._castModels.push(model); }
+            if (model.enabled && model.node && model.castShadow && (model.node && ((visibility & model.node.layer) === model.node.layer))) { this._castModels.push(model); }
```
Comment (minggo) [inline cocos/rendering/planar-shadow-queue.ts]: It is better to have a different line.

File: native/cocos/renderer/pipeline/PlanarShadowQueue.cpp [hunk 4]
```diff
@@ -42,7 +40,8 @@ namespace cc {
 namespace pipeline {
 
 PlanarShadowQueue::PlanarShadowQueue(RenderPipeline *pipeline)
-: _pipeline(pipeline) {
+: _phaseID(getPhaseID("planar-shadow")) {
+    _pipeline = pipeline;
```
Comment (minggo) [inline native/cocos/renderer/pipeline/PlanarShadowQueue.cpp]: Why initialize _pipeline here?

Comment (troublemaker52025) [reply native/cocos/renderer/pipeline/PlanarShadowQueue.cpp]: A weak pointer reference to _pipeline is required. The following section needs to get the pipelineSceneData from _pipeline

Comment (minggo) [reply native/cocos/renderer/pipeline/PlanarShadowQueue.cpp]: Why can not initialize it in parameter list?

**Review by minggo — APPROVED** [review]

File: native/cocos/renderer/pipeline/PlanarShadowQueue.cpp (hunk 4 above)
Comment (troublemaker52025) [reply native/cocos/renderer/pipeline/PlanarShadowQueue.cpp]: > 

Referring to the ShadowMapBatchedQueue.cpp file, I wanted to be consistent in the style of my code.

**Pull request merged**